### DISCUSSION TITLE
How to show attachments of SP list items in custom results template?

### DISCUSSION THREAD

**Discussion by anwimafi** [microsoft-search/pnp-modern-search#4082] [Q&A] [answered]
Hello, 
in my organisation we are running an adaption (the separate wepparts habe been combined to a single one) of pnp-modern search. Since the basics are the same, I hope to find some help here. 
To improve user experience I am templating the result layouts, for example creating own tile layouts for page searches. That works pretty well and I was able to apply the needed handlebars. So far so good. 
Now I would like to search a list (actually to pull the whole list) and display its contents in a user requested way. The elements in SP lists have the possibility to include attchements (i.e. files). Regarding my current experience with templating the result set I would have expected a managed property, maybe containing an array or list of the attached files. The only "attachment named" properties I could find in the webpart are AttachmentURI and AttachmentType. But both do return null. 
So, is it possible to show some list item attachement based data in pnp-mysearch template, and if yes, how would it work?
Thanks in advance

**Comment by kasperbolarsen** [comment, marked as answer]
using SP Editor ( the smartest thing for a Search consultant since sliced bread) I can see that you should take a look at 
![image](https://github.com/user-attachments/assets/7692698d-b8bc-45f3-a86e-ae17f3c2f19c)

**Reply by anwimafi** [reply]
Many thanks! That is exactly what I was looking for. 
SP Editor seems to be a great tool, unfortunately it is blocked here.

**Reply by kasperbolarsen** [reply]
If browser extensions are blocked you might find the SharePoint Search Query tool useful, https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/
